src/ncp/ncp_spinel.cpp

```diff
@@ -257,55 +257,39 @@ void NcpSpinel::HandleResponse(spinel_tid_t aTid, const uint8_t *aFrame, uint16_
 
     SuccessOrExit(error = SpinelDataUnpack(aFrame, aLength, kSpinelDataUnpackFormat, &header, &cmd, &key, &data, &len));
 
-    VerifyOrExit(cmd == SPINEL_CMD_PROP_VALUE_IS && key == mWaitingKeyTable[aTid], error = OTBR_ERROR_INVALID_STATE);
+    VerifyOrExit(cmd == SPINEL_CMD_PROP_VALUE_IS, error = OTBR_ERROR_INVALID_STATE);
 
-    switch (key)
+    switch (mCmdTable[aTid])
     {
-    case SPINEL_PROP_LAST_STATUS:
+    case SPINEL_CMD_PROP_VALUE_SET:
     {
-        spinel_status_t status = SPINEL_STATUS_OK;
-
-        failureHandler = [this, aTid](otError aError) { HandleResponseForCommand(aTid, aError); };
-        SuccessOrExit(error = SpinelDataUnpack(data, len, SPINEL_DATATYPE_UINT_PACKED_S, &status));
-
-        HandleResponseForCommand(aTid, ot::Spinel::SpinelStatusToOtError(status));
+        error = HandleResponseForPropSet(aTid, key, data, len);
         break;
     }
-    case SPINEL_PROP_THREAD_ACTIVE_DATASET:
-    {
-        CallAndClear(mDatasetSetActiveTask, OT_ERROR_NONE);
-        break;
-    }
-    case SPINEL_PROP_NET_IF_UP:
-    {
-        CallAndClear(mIp6SetEnabledTask, OT_ERROR_NONE);
-        break;
-    }
-    case SPINEL_PROP_NET_STACK_UP:
+    case SPINEL_CMD_NET_CLEAR:
     {
-        CallAndClear(mThreadSetEnabledTask, OT_ERROR_NONE);
+        spinel_status_t status = SPINEL_STATUS_OK;
+
+        SuccessOrExit(error = SpinelDataUnpack(data, len, SPINEL_DATATYPE_UINT_PACKED_S, &status));
+        CallAndClear(mThreadErasePersistentInfoTask, ot::Spinel::SpinelStatusToOtError(status));
         break;
     }
     default:
         break;
     }
 
 exit:
-    FreeTid(aTid);
-
     if (error == OTBR_ERROR_INVALID_STATE)
     {
-        otbrLogCrit("Received unexpected response with cmd:%u, key:%u, waiting key:%u for tid:%u", cmd, key,
-                    mWaitingKeyTable[aTid], aTid);
+        otbrLogCrit("Received unexpected response with (cmd:%u, key:%u), waiting (cmd:%u, key:%u) for tid:%u", cmd, key,
+                    mCmdTable[aTid], mWaitingKeyTable[aTid], aTid);
     }
     else if (error == OTBR_ERROR_PARSE)
     {
         otbrLogCrit("Error parsing response with tid:%u", aTid);
-        if (failureHandler)
-        {
-            failureHandler(OtbrErrorToOtError(error));
-        }
```

Comment:
This block calling `failureHandler` seems to be removed? Is this intentional?

Reply:
Yes, it's intentional. My idea is that both `INVALID_STATE` and `PARSE` errors are unexpected and users (the caller of NcpSpinel async APIs) cannot handle such errors. When theses errors happened, that means the stack has run into some problems. It's unlike the case where user calls a NcpSpinel API, and the API returns an error such as `INVALID_STATE` through the callback because that's the result of the operation. So I think we probably don't return errors through callback to the caller in these cases.